lib/view_model/active_record/update_operation.rb

```diff
@@ -131,33 +129,36 @@ def run!(deserialize_context:)
         # deletes are handled elsewhere.
 
         changes = viewmodel.changes
-
         if changes.new? || changes.changed_attributes.present? || changes.changed_associations.present?
+          debug "-> #{debug_name}: Validating changes"
+          # invoke pre-save hook
           viewmodel.before_save(changes, deserialize_context: deserialize_context)
 
-          # The hook before this might have caused additional changes. All changes should be checked by the policy,
-          # so we have to recalculate the change set.
-
-          changes = viewmodel.changes
+          # Invoke any validation hooks before attempting to save.
+          viewmodel.validate!
 
-          deserialize_context.editable!(viewmodel,
-                                        initial_editability: initial_editability,
-                                        changes: changes)
+          # The hooks before this might have caused additional changes. All
+          # changes should be checked by the policy, so we have to recalculate
+          # the change set. Note that changes to associated children that point
+          # to this parent are recorded in `changes`, but their models have not
+          # yet been updated.
+          final_changes = viewmodel.changes
+          deserialize_context.run_callback(ViewModel::Callbacks::Hook::OnChange, viewmodel, changes: final_changes)
```

Comment:
Can we `raise` if `changes` changed here?

Reply:
where?